src/fifo/FifoControllerIndirect.cpp

```diff
@@ -24,6 +24,7 @@ FifoControllerIndirect::FifoControllerIndirect(uint32_t numFrames,
                                                int64_t * readCounterAddress,
                                                int64_t * writeCounterAddress)
         : FifoControllerBase(numFrames, threshold)
+        // Why are we casting `int64_t *` to `std::atomic<uint64_t> *`?
```

Comment:
Maybe we should change the API for this class to take reference to atomics.

Reply:
Bug #589

Reply:
Fixed in #603, (Specifically 08a5e0ff6c33cd6ab9ad241af12f9b068d9315cd)